enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,224 @@
+import { renderWithProviders, screen } from "__support__/ui";
+import { getNextId } from "__support__/utils";
+import type { Database, Table } from "metabase-types/api";
+import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { DataSourceSelector } from "../DataSelector";
+
+const DATABASES = [createSampleDatabase()];
+
+const storeInitialState = createMockState({
+  settings: createMockSettingsState({
+    "enable-nested-queries": true,
+  }),
+});
+
+const AVAILABLE_MODELS: Record<AvailableModels, ("dataset" | "table")[]> = {
+  "tables-only": ["table"],
+  "tables-and-models": ["dataset", "table"],
+};
+
+type AvailableModels = "tables-only" | "tables-and-models";
+
+interface SetupOpts {
+  databases?: Database[];
+  isJoinStep?: boolean;
+  availableModels?: AvailableModels;
+  selectedTable?: {
+    id: number | undefined;
+    databaseId: number | null;
+  };
+}
+
+function setup({
+  databases = DATABASES,
+  isJoinStep = false,
+  availableModels = "tables-only",
+  selectedTable = {
+    id: undefined,
+    databaseId: null,
+  },
+}: SetupOpts = {}) {
+  fetchMock.get(
+    {
+      url: "path:/api/search",
+      query: {
+        calculate_available_models: true,
+        limit: 0,
+        models: ["dataset"],
+      },
+    },
+    {
+      data: [],
+      limit: 0,
+      models: ["dataset"],
+      offset: 0,
+      table_db_id: null,
+      engine: "search.engine/in-place",
+      total: 1,
+      available_models: AVAILABLE_MODELS[availableModels],
+    },
+  );
+
+  setupDatabasesEndpoints(databases, undefined, { saved: true });
+
+  return renderWithProviders(
+    <DataSourceSelector
+      isInitiallyOpen
+      canChangeDatabase={!isJoinStep}
+      selectedDatabaseId={selectedTable.databaseId}
+      selectedTableId={selectedTable.id}
+      canSelectModel={true}
+      canSelectTable={true}
+      triggerElement={<div>Click me to open or close data picker</div>}
+      setSourceTableFn={jest.fn()}
+    />,
+    { storeInitialState },
+  );
+}
+
+describe("DataSourceSelector", () => {
+  describe("only tables are available", () => {
+    const setupOpts: SetupOpts = {
+      availableModels: "tables-only",
+    };
+
+    it("should close the picker when clicking outside", async () => {
+      setup(setupOpts);
+      expect(await screen.findByText("Orders")).toBeInTheDocument();
+      expect(screen.getByText("Sample Database")).toBeInTheDocument();
+
+      await userEvent.click(
+        screen.getByText("Click me to open or close data picker"),
+      );
+
+      expect(screen.queryByText("Orders")).not.toBeInTheDocument();
+      expect(screen.queryByText("Sample Database")).not.toBeInTheDocument();
+    });
+
+    it("should show search input when there are 10 and more tables", async () => {
```

Comment:
this is one of the edge cases I found. If we find more of these cases in the future, we could add more tests here.

Here's where it's implemented
https://github.com/metabase/metabase/blob/dff0c24b0bdff1286d58e65162e4072e24f6116e/enterprise/frontend/src/embedding/data-picker/DataSelectorTablePicker/DataSelectorTablePicker.tsx#L55